### PR TITLE
8211002: test/jdk/java/lang/Math/PowTests.java skips testing for non-corner-case values

### PR DESCRIPTION
**Pull request by bplb**
Please consider this proposal to add some test coverage comparing `Math` and `StrictMath` results of `pow()`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8211002](https://bugs.openjdk.java.net/browse/JDK-8211002): test/jdk/java/lang/Math/PowTests.java skips testing for non-corner-case values

### Reviewers
 * [Joe Darcy](https://openjdk.java.net/census#darcy) (@jddarcy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4758/head:pull/4758` \
`$ git checkout pull/4758`

Update a local copy of the PR: \
`$ git checkout pull/4758` \
`$ git pull https://git.openjdk.java.net/jdk pull/4758/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4758`

View PR using the GUI difftool: \
`$ git pr show -t 4758`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4758.diff">https://git.openjdk.java.net/jdk/pull/4758.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/Math/PowTests.java [hunk 1]
```diff
@@ -57,6 +57,21 @@ static int testNonstrictPowCase(double input1, double input2, double expected) {
         return failures;
     }
 
+    static int testStrictVsNonstrictPowCase(double input1, double input2) {
+        double smResult = StrictMath.pow(input1, input2);
+        double mResult = Math.pow(input1, input2);
+        int failures = 0;
+        if (Double.isNaN(smResult)) {
+            if (!Double.isNaN(mResult)) failures = 1;
```
Comment (jddarcy) [inline test/jdk/java/lang/Math/PowTests.java:65]: Pretty sure the Tests.testUlpDiff call will handle two NaNs as well as NaN/non-NaN argument.

File: test/jdk/java/lang/Math/PowTests.java [hunk 2]
```diff
@@ -57,6 +57,21 @@ static int testNonstrictPowCase(double input1, double input2, double expected) {
...
+        } else {
+            failures += Tests.testUlpDiff(
+                "StrictMath.pow(double, double) vs Math.pow(double, double)",
+                input1, input2, mResult, smResult, 1.0
```
Comment (jddarcy) [inline test/jdk/java/lang/Math/PowTests.java:69]: In theory at least, one pow method could round an ulp up and the other could round an ulp down so the difference should be 2 ulps (ignoring the special case of values straddling a power of two).

File: test/jdk/java/lang/Math/PowTests.java (hunk 1 above)
Comment (bplb) [reply test/jdk/java/lang/Math/PowTests.java]: Looks like that is true.

File: test/jdk/java/lang/Math/PowTests.java (hunk 2 above)
Comment (bplb) [reply test/jdk/java/lang/Math/PowTests.java]: Good point.

**Comment by bplb** [comment]
Ping.

**Review by jddarcy — APPROVED** [review]

**Pull request closed without merging**